### PR TITLE
Add a Calendar to avoid downloading old news

### PR DESCRIPTION
**Pull request by RachidTagzen**
I added a calendar to avoid downloading old news to preserve a small database.
I hope that you add it to the official release and improve it.
Thank you.

### CODE REVIEW

File: src/feedpropertiesdialog.cpp [hunk 1]
```diff
@@ -19,624 +19,678 @@
 #include "feedpropertiesdialog.h"
 
 FeedPropertiesDialog::FeedPropertiesDialog(bool isFeed, QWidget *parent)
-  : Dialog(parent)
-  , isFeed_(isFeed)
+    : Dialog(parent)
```
Comment (genodeftest) [inline src/feedpropertiesdialog.cpp]: These whitespace changes are not related to the merge request, are they? It would be nicer and cleaner to not have them then.

Comment (lmofallis) [reply src/feedpropertiesdialog.cpp]: These whitespace are made by the original developer not me.

Comment (genodeftest) [reply src/feedpropertiesdialog.cpp]: GitHub claims those whitespace changes have been done by you. Maybe your editor has auto-formatting enabled?

Comment (RachidTagzen) [reply src/feedpropertiesdialog.cpp]: They are not.

Comment (genodeftest) [reply src/feedpropertiesdialog.cpp]: I do not understand. There is a diff (by you) and it clearly states that the whitespace changes. Who should it be if not you who added that whitespace??

Comment (RachidTagzen) [reply src/feedpropertiesdialog.cpp]: Oddly, perhaps my Qt Creator added them, but not me.

File: lang/quiterss_ar.ts [hunk 2]
```diff
@@ -527,7 +529,7 @@
     </message>
     <message>
         <location filename="../src/cleanupwizard.cpp" line="361"/>
-        <source>Totally remove records that had marked 'deleted' from DB.
+        <source>Totally remove records that had marked &apos;deleted&apos; from DB.
```
Comment (genodeftest) [inline lang/quiterss_ar.ts:532]: Probably more automatic changes done by your IDE which bloat up the diff. Looks like all translation files are affected.

File: src/database/database.cpp [hunk 3]
```diff
@@ -29,434 +29,462 @@
...
-    "disableUpdate integer default 0, "     // disable update feed
-    "javaScriptEnable integer default 1, "  //
-    // version 16
-    "layoutDirection integer default 0, "    // 0 - ltr; 1 - rtl
-    // Version 17
-    "SingleClickAction integer default 0, " // ENewsClickAction
-    "DoubleClickAction integer default 0, " // ENewsClickAction
-    "MiddleClickAction integer default 0 "  // ENewsClickAction
-    ")");
+        "CREATE TABLE feeds("
```
Comment (genodeftest) [inline src/database/database.cpp:32]: More needless whitespace changes.

File: lang/quiterss_ar.ts (hunk 2 above)
Comment (RachidTagzen) [reply lang/quiterss_ar.ts:532]: I think so saddly.

File: src/feedpropertiesdialog.cpp (hunk 1 above)
Comment (genodeftest) [reply src/feedpropertiesdialog.cpp]: It seems to affect other files too. A diff of >50k lines is not expected for such a comparably small change.

File: src/database/database.cpp (hunk 3 above)
Comment (RachidTagzen) [reply src/database/database.cpp:32]: Ok. In your opinion, what's the best way to fix this automatic unwanted changes?

File: src/feedpropertiesdialog.cpp (hunk 1 above)
Comment (RachidTagzen) [reply src/feedpropertiesdialog.cpp]: I totaly agree with you.

File: src/database/database.cpp (hunk 3 above)
Comment (genodeftest) [reply src/database/database.cpp:32]: I would use a Git frontend (I prefer GNOME Gitg) combined with the console and follow these steps (roughly):
1. Create a backup branch (`git branch backup-branch`)
2. `git reset HEAD~5` to unpush the latest changes from your git history. File changes are still preserved.
3. use the git gui (or command line tool) to revert the needless whitespace and encoding changes. Command line: `git checkout -- [filename]`
4. Copy commit messages from your "old" commits

For the future, I would recommend disabling auto-formatting the code in your editor/IDE. Yeah, that's quite annoying.

Comment (genodeftest) [reply src/database/database.cpp:32]: With `git push --force` you should be able to push the modified changes back to this pull request.

Comment (genodeftest) [reply src/database/database.cpp:32]: I like the idea of this pull request, it is just hard to have a look at the code if the diff is cluttered by so much noise.

Comment (RachidTagzen) [reply src/database/database.cpp:32]: Thank you very much.
It's my first time to share a code in github, and I don't expect this issues.
I will try to fix this ASAP.

Comment (genodeftest) [reply src/database/database.cpp:32]: Feel free to ask for more help and welcome to the contribution world ;)
